common/src/types.rs

```diff
@@ -0,0 +1,125 @@
+use crate::crypto::*;
+use serde::{Deserialize, Serialize};
+
+pub type VotingPower = u64;
+/// A UNIX timestamp measured in milliseconds.
+pub type Timestamp = u64;
+/// A block height. The genesis block is at height 0.
+pub type BlockHeight = u64;
+pub type ConsensusRound = u64;
+pub type FinalizationProof = Vec<(PublicKey, TypedSignature<BlockHeader>)>;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+pub struct Member {
+    pub public_key: PublicKey,
+    pub name: String,
+    pub governance_voting_power: u64,
+    pub consensus_voting_power: u64,
+    pub governance_delegations: Option<PublicKey>,
+    pub consensus_delegations: Option<PublicKey>,
+    // TODO: add various conditions for each delegation.
+    // - Unlock-Automatically-After-N-Blocks
+    // - Unlock-Automatically-After-T-Seconds
+    // - Unlock-If-The-Delegatee-Is-Not-Active
+    // - Unlock-If-The-Validator-Set-Changes
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+pub struct BlockHeader {
```

Comment:
Is `delegation_state_hash` not necessary anymore? Why was it thought of as needed at the first time?

Reply:
it is changed as a design choice, after the huge update on the state architecture.

Reply:
Can you show me the PR where the state architecture is updated?

Reply:
We introduced 'Git blockchain state'. That's what I meant.

Reply:
Got it.